src/ssl/PeekingPeerConnector.h

```diff
@@ -70,6 +70,10 @@ class PeekingPeerConnector: public Security::PeerConnector {
     AsyncCall::Pointer closeHandler; ///< we call this when the connection closed
     bool splice; ///< whether we are going to splice or not
     bool serverCertificateHandled; ///< whether handleServerCertificate() succeeded
+
+    /// The bumping mode at the time the first PeekingPeerConnector attempt
+    /// is started
+    Ssl::BumpMode currentBumpMode = Ssl::bumpNone;
```

Comment:
This description implies that we _preserve_ the bumping mode (whatever that is) computed during the first PeekingPeerConnector attempt. However, we never preserve that computation value! Instead, we always compute it from scratch for each PeekingPeerConnector attempt. Thus, the description of this field has to match the "assembled" description derived from the corresponding initialization code -- bumpModeAtStep(2) -- currently "the bumping mode at step2".

If the above logic is correct, then this data member is mostly a performance optimization -- we simply do not want to call bumpModeAtStep(2) multiple times (or worry about the client going away) and cache the computed (essentially constant) result. There is nothing related to multiple PeekingPeerConnector attempts here!

I can adjust/update the description, but I want to make sure I am not missing something. I am worried that you wanted to talk about multiple PeekingPeerConnector attempts here, and I do not know why.

Reply:
> This description implies that we _preserve_ the bumping mode (whatever that is) computed during the first PeekingPeerConnector attempt. However, we never preserve that computation value! Instead, we always compute it from scratch for each PeekingPeerConnector attempt. Thus, the description of this field has to match the "assembled" description derived from the corresponding initialization code -- bumpModeAtStep(2) -- currently "the bumping mode at step2".

Yes you are right. 
Probably we should also rename this member to something like bumpModeAtStep2. Does this sounds good?

> 
> If the above logic is correct, then this data member is mostly a performance optimization -- we simply do not want to call bumpModeAtStep(2) multiple times (or worry about the client going away) and cache the computed (essentially constant) result. There is nothing related to multiple PeekingPeerConnector attempts here!

Yes.

> 
> I can adjust/update the description, but I want to make sure I am not missing something. I am worried that you wanted to talk about multiple PeekingPeerConnector attempts here, and I do not know why.

No, this is definitely a wrong description. Probably I just wanted to write "at the time the PeekingPeerConenctor attempt is started". For a reason the word "first" was badly placed here.

Reply:
Aha, thank you for clarifying this! I do not think bumpModeAtStep2 is the best choice because step2 may be skipped. How about something like this:

```C++
/// An (explicit or implicit) ssl_bump action that led to our job creation.
/// This is usually a step2 peek or stare action, but it may be step1 server-first.
/// Caching that hard-to-compute action ID simplifies and optimizes our code.
const Ssl::BumpMode lastSslBumpAction;
```

And add a `static Ssl::BumpMode LastSslBumpActionExtractor(const HttpRequest &)` method that our constructor can call to set this constant data member at member initialization time? The method is static because we do not want to worry about accidentally accessing a not-yet-initialized PeekingPeerConnector data member while computing the return value.

Reply:
commit 29394fe